mono/mini/interp/interp.c

```diff
@@ -6651,6 +6653,41 @@ register_interp_stats (void)
 	mono_counters_register ("Inline failures", MONO_COUNTER_INTERP | MONO_COUNTER_INT, &mono_interp_stats.inline_failures);
 }
 
+static const MonoEECallbacks mono_interp_callbacks = {
+#ifdef MONO_ARCH_HAVE_INTERP_ENTRY_TRAMPOLINE
+	interp_entry_from_trampoline,
+#else
+	NULL,
+#endif
+	interp_to_native_trampoline,
+	interp_create_method_pointer,
+	interp_create_method_pointer_llvmonly,
+	interp_runtime_invoke,
+	interp_init_delegate,
+	interp_delegate_ctor,
+	interp_get_remoting_invoke,
+	interp_set_resume_state,
+	interp_run_finally,
+	interp_run_filter,
+	interp_frame_iter_init,
+	interp_frame_iter_next,
+	interp_find_jit_info,
+	interp_set_breakpoint,
+	interp_clear_breakpoint,
+	interp_frame_get_jit_info,
+	interp_frame_get_ip,
+	interp_frame_get_arg,
+	interp_frame_get_local,
+	interp_frame_get_this,
+	interp_frame_arg_to_data,
+	interp_data_to_frame_arg,
+	interp_frame_arg_to_storage,
+	interp_frame_arg_set_storage,
+	interp_frame_get_parent,
+	interp_start_single_stepping,
+	interp_stop_single_stepping,
```

Comment:
I'm actually not a fan of this. What we had previously, gave us some "extra safety".  Here it's easy to get the ordering wrong.

Reply:
It is nice to match the names up I agree.
 Structs in general cannot be edited lightly because of this sensitivity.
Designated initializers in C++ would help with this.
Readonlyness and static initialization are also valuable. You can't have everything.
Function pointers are sercurity attack vectors and read only ones are preferred.
Granted you can overwrite the data too.
As well as initializing all the fields -- though you could memset to to get that.
The original had uninitialized values.

Notice that typesafety does help, some.
The compiler won't let you initialize functoin pointers w/ functions whose signature does not match. But the signatures are not necesssarily varying, and some could match others.

Reply:
Also despite the ordering requirement, this is a heavily used idiom.

Reply:
I agree, having uninitialized values for the stub case is problematic. Please submit a PR for that, but changing the structure of the code just because of that is not the way to go. This comes to my mind: https://tirania.org/blog/archive/2010/Dec-31.html

We have `MonoDebuggerCallbacks`, `MonoUtilityThreadCallbacks`, `MonoThreadInfoRuntimeCallbacks`, `MonoThreadInfoCallbacks`, `MonoCodeManagerCallbacks`, `MonoRuntimeCallbacks`, `MonoRuntimeExceptionHandlingCallbacks` and others that all follow the same pattern as `MonoEECallbacks`, so I'm reluctant to change that.  Please don't understand this as a call to change the others.

Reply:
I was already intending to them them all, just one at a time.
The data really should be readonly.